File: sdks/python/examples/market-data/recent_trades.py
import pmxt

def main():
    # Kalshi
    kalshi = pmxt.Kalshi()
    k_event = kalshi.fetch_events(query="Fed Chair")[0]
    k_market = k_event.search_markets("Kevin Warsh")[0]

    k_trades = kalshi.fetch_trades(k_market.yes.id, pmxt.HistoryFilterParams(resolution='1h', limit=10))
    print('Kalshi:', k_trades)

    # Polymarket
    poly = pmxt.Polymarket()
    p_event = poly.fetch_events(query='Fed Chair')[0]
    p_market = p_event.search_markets('Kevin Warsh')[0]

    p_trades = poly.fetch_trades(p_market.yes.id, pmxt.HistoryFilterParams(resolution='1h', limit=10))
    print('Polymarket:', p_trades)

if __name__ == "__main__":
    main()

File: sdks/typescript/examples/market-data/recent_trades.ts
import pmxt from 'pmxtjs';

const main = async () => {
    // Kalshi
    const kalshi = new pmxt.Kalshi();
    const kEvent = (await kalshi.fetchEvents({ query: 'Fed Chair' }))[0];
    const kMarket = kEvent.searchMarkets('Kevin Warsh')[0];

    const kTrades = await kalshi.fetchTrades(kMarket.yes!.id, { limit: 10, resolution: '1m' });
    console.log('Kalshi:', kTrades);

    // Polymarket
    const poly = new pmxt.Polymarket();
    const pEvent = (await poly.fetchEvents({ query: 'Fed Chair' }))[0];
    const pMarket = pEvent.searchMarkets('Kevin Warsh')[0];

    // Use .yes.id for convenience
    const pTrades = await poly.fetchTrades(pMarket.yes!.id, { limit: 10, resolution: '1m' });
    console.log('Polymarket:', pTrades);
};

main();